src/Model/List/SS_List.php

```diff
@@ -15,83 +16,243 @@
+     */
+    public function excludeAny(...$args): SS_List;
+
+    /**
+     * Return a new instance of this list that excludes any items with these characteristics
+     * Filter this List by a callback function. The function will be passed each record of the List in turn,
+     * and must return true for the record to be included. Returns the filtered list.
+     *
+     * @example $list = $list->filterByCallback(function($item, $list) { return $item->Age == 9; })
+     * @return SS_List<T>
+     */
+    public function filterByCallback(callable $callback): SS_List;
+
+    /**
+     * Return the first item with the given ID
+     *
+     * @param int $id
+     * @return T|null
+     */
+    public function byID(int|string|null $id): mixed;
+    // Note that string ID's and null values should be handled as things like
+    // form submissions and controller params will often be string IDs,
+    // and null is often unintentionally passed in a lot of instances.
+    // If this is changed in the future to only accept int, ensure that all other
+    // filtering methods are updated at the same time for consistency
+
+    /**
+     * Filter this list to only contain the given Primary IDs
+     *
+     * @param array $ids Array of integers
+     * @return SS_List<T>
+     */
+    public function byIDs(array $ids): SS_List;
+
+    /**
+     * Returns TRUE if the list can be sorted by a field.
+     */
+    public function canSortBy(string $by): bool;
+
+    /**
+     * Return a new instance of this list that is sorted by one or more fields. You can either pass in a single
+     * field name and direction, or a map of field names to sort directions.
+     *
+     * @example $list = $list->sort('Name'); // default ASC sorting
+     * @example $list = $list->sort('Name DESC'); // DESC sorting
+     * @example $list = $list->sort('Name', 'ASC');
+     * @example $list = $list->sort(array('Name'=>'ASC,'Age'=>'DESC'));
+     *
+     * @return SS_List<T>
+     */
+    public function sort(...$args): SS_List;
+
+
+    /**
+     * Return a new instance of this list based on reversing the current sort.
+     *
+     * @example $list = $list->reverse();
+     *
+     * @return SS_List<T>
+     */
+    public function reverse(): SS_List;
+
+    /**
+     * Returns a new instance of this list where no more than $limit records are included.
+     * If $offset is specified, then that many records at the beginning of the list will be skipped.
+     * This matches the behaviour of the SQL LIMIT clause.
+     *
+     * If `$length` is null, then no limit is applied. If `$length` is 0, then an empty list is returned.
+     *
+     * @throws InvalidArgumentException if $length or offset are negative
+     * @return SS_List<T>
+     */
+    public function limit(?int $length, int $offset = 0): SS_List;
+
+    /**
+     * Returns an Iterator for this list
+     *
+     * @return Traversable<T>
+     */
+    public function getIterator(): Traversable;
```

Comment:
This is part of the `IteratorAggregate` interface already
```suggestion
```

Reply:
Removed